### PR TITLE
chore: Ensure `Profiling.prepare` is called before profiling

### PR DESCRIPTION
**Pull request by iamrajiv**
Fixes: https://github.com/dispatchrun/wzprof/issues/93

### CODE REVIEW

File: wzprof.go [hunk 1]
```diff
@@ -147,9 +137,31 @@ func (p *Profiling) Prepare(mod wazero.CompiledModule) error {
 		}
 		p.symbols = buildDwarfSymbolizer(dwarf)
 	}
+
+	// Set the flag to true if Prepare succeeds
+	p.prepareCalled = true
+
 	return nil
 }
```
Comment (achille-roussel) [inline wzprof.go:145]: There's a severe usability impact by breaking the API and having to handle errors in a lot more contexts here. It feels like in an attempt to improve usability (by automatically preparing the module) we end up in a worse situation. 

How about ensuring Prepare is called by panicking if profilers are created before? A clear error message can go a long way and we could keep the APIs unchanged and free of error handling.

File: cmd/wzprof/main.go [hunk 2]
```diff
@@ -78,25 +92,6 @@ func (prog *program) run(ctx context.Context) error {
 		}
 	}
 
-	ctx = context.WithValue(ctx,
```
Comment (achille-roussel) [inline cmd/wzprof/main.go:81]: Why did we need to remove this?

**Review by achille-roussel — APPROVED** [review]
Thanks for the contribution!

**Comment by achille-roussel** [comment]
@iamrajiv do you mind taking a look at the linter failures?

The build failure seems caused by a missing dependency, let's not worry about it here.

**Comment by iamrajiv** [comment]
Yeah, @achille-roussel, I created a PR to fix the linter issues i.e. https://github.com/dispatchrun/wzprof/pull/105.

**Pull request merged**